python/lsst/pipe/tasks/calibrate.py

```diff
@@ -453,13 +469,22 @@ def run(self, exposure, defects=None, idFactory=None):
         self.initialMeasurement.run(exposure, sources1, allowApCorr=False)
 
         if self.config.doPsf:
+            matches = None
             if self.config.doAstrometry:
-                astromRet = self.astrometry.run(exposure, sources1)
-                matches = astromRet.matches
-            else:
                 # If doAstrometry is False, we force the Star Selector to either make them itself
                 # or hope it doesn't need them.
-                matches = None
+                origWcs = exposure.getWcs()
+                try:
+                    astromRet = self.astrometry.run(exposure, sources1)
+                    matches = astromRet.matches
+                except RuntimeError as e:
+                    if self.config.requireAstrometry:
+                        raise
+                    self.log.warn("Unable to perform astrometry (%s): attempting to proceed" % e)
+                finally:
+                    # Restore original Wcs: we're going to repeat the astrometry later, and if it succeeded
+                    # this time, running it again with the same basic setup means it should succeed again.
+                    exposure.setWcs(origWcs)
```

Comment:
No, we want to solve the astrometry (so `forceKnownWcs=False`), but we want to undo it when we're done.